### PR TITLE
*: bump go version to 1.11

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Go1.11 now release, make TiDB to use the latest Go.

### What is changed and how it works?

Update circle.yml file to use Go1.11.
Fix `make lint` errors.
Fix `make vet` errors.

### Check List 

Tests 

 - No code

Code changes

Mainly due to lint and vet complain.

Side effects

 - After this change, TiDB requires Go1.11

### CODE REVIEW

**Comment by tiancaiamao** [comment]
Oh, build fail. It has to wait for the circle go1.11 image.

File: Makefile [hunk 1]
```diff
@@ -117,7 +117,7 @@ lint:
 
 vet:
 	@echo "vet"
-	@retool do govet -all -shadow $$($(PACKAGE_DIRECTORIES)) 2>&1 | $(FAIL_ON_STDOUT)
+	@go vet -all -shadow $(PACKAGES) 2>&1 | $(FAIL_ON_STDOUT)
```
Comment (winoros) [inline Makefile]: https://github.com/pingcap/tidb/pull/7197
It's changed in this pr. Should we change it back?

Comment (tiancaiamao) [reply Makefile]: IMHO, the introduce of `retool` in https://github.com/pingcap/tidb/pull/7197 is good for project management, it makes TiDB survive from third-party tool changes, so we'd better keep it. @winoros 

I don't know why `govet` fail here under the latest Go1.11, but the standard go vet tool works, this change is a workaround.

**Comment by coocood** [comment]
LGTM

**Pull request merged**